Kernel/ProcessExposed.cpp

```diff
@@ -246,4 +168,83 @@ ErrorOr<void> ProcFSExposedDirectory::traverse_as_directory(FileSystemID fsid, F
     return {};
 }
 
+class ProcFSSelfProcessDirectory final : public ProcFSExposedLink {
```

Comment:
Am I missing something or where does this new handler for `/proc/self` come from?

Reply:
It came from the `GlobalProcessExposed.cpp` file:
https://github.com/SerenityOS/serenity/blob/2079728a7455c7c2d852c2784f7b02bcf16f3583/Kernel/GlobalProcessExposed.cpp#L383-L393
I delete that file but I still want to preserve the `self` directory because it's important.